agent/util-scripts/pbench-tool-meister-start

```diff
@@ -282,34 +284,86 @@ def main(argv):
         logger.exception("failed to load tool group data")
         return 1
 
+    # See if anybody told us to use certain options with SSH commands.
+    ssh_opts = os.environ.get("ssh_opts", "")
+
     try:
         benchmark_run_dir = os.environ["benchmark_run_dir"]
         hostname = os.environ["_pbench_hostname"]
         full_hostname = os.environ["_pbench_full_hostname"]
     except Exception:
         logger.exception("failed to fetch parameters from the environment")
-        return 1
+        return 2
     else:
         tm_dir = Path(benchmark_run_dir, "tm")
         try:
             tm_dir.mkdir()
             os.chdir(tm_dir)
         except Exception:
             logger.exception("failed to create the local tool meister directory")
-            return 1
+            return 3
     if not full_hostname or not hostname:
         logger.error(
             "ERROR - _pbench_hostname ('%s') and _pbench_full_hostname ('%s')"
             " environment variables are required",
             hostname,
             full_hostname,
         )
-        return 1
-    if os.environ.get("_PBENCH_UNIT_TESTS"):
-        # FIXME: this is an artifact of the unit test environment.
-        hostnames = "localhost"
+        return 4
+
+    # Determine the Tool Meister "hostname" to use for the Redis server to
+    # bind to, in addition to "localhost". That same host name will be used by
+    # the Tool Data Sink to bind to as well.  If the caller's environment
+    # contains a PBENCH_TM_BIND_HOSTNAME environment variable, we'll use that,
+    # if not, we'll use the value from the _pbench_full_hostname environment
+    # variable.
+    tm_bind_hostname = os.environ.get("PBENCH_TM_BIND_HOSTNAME", full_hostname)
+    hostnames_l = []
+    try:
+        localhost_ip = socket.gethostbyname("localhost")
+    except socket.error:
+        # Interesting networking environment, no IP address for "localhost" ...
+        localhost_ip = None
     else:
-        hostnames = f"localhost {full_hostname}"
+        # Add to the list of host names the Redis server will use.
+        hostnames_l.append("localhost")
+    try:
+        tm_bind_hostname_ip = socket.gethostbyname(tm_bind_hostname)
```

Comment:
This won't be enough, unfortunately.  It's possible that what gethostbyname returns isn't the actual IP address.  I really need it to be possible to completely disassociate `gethostbyname` from what redis-server binds to.

Note that on some pods I'm able to resolve my controller IP address:

```
sh-4.4# host pbench-controller.pbench
pbench-controller.pbench.svc.cluster.local has address 172.30.157.50
sh-4.4# exit
```

While on others I can't:

```[kni@e23-h05-740xd T]$ oc get pods
NAME                                 READY   STATUS    RESTARTS   AGE
pbench-agent-host-788dfc6b67-72htb   1/1     Running   0          4m37s
pbench-agent-host-788dfc6b67-xnhtn   1/1     Running   0          4m37s
pbench-client-1                      1/1     Running   0          4m38s
pbench-controller                    1/1     Running   0          4m37s
pbench-server-1                      1/1     Running   0          4m38s

[kni@e23-h05-740xd T]$ oc rsh pbench-agent-host-788dfc6b67-xnhtn
sh-4.4# host pbench-controller.pbench.svc.cluster.local
Host pbench-controller.pbench.svc.cluster.local not found: 3(NXDOMAIN)
```

And on the controller itself what it resolves to is not the same as the actual IP address:

```
[kni@e23-h05-740xd T]$ oc rsh pbench-controller
sh-4.4# host pbench-controller.pbench.svc.cluster.local
pbench-controller.pbench.svc.cluster.local has address 172.30.157.50
sh-4.4# ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
3: eth0@if220: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc noqueue state UP group default 
    link/ether 0a:58:0a:83:00:c2 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.131.0.194/23 brd 10.131.1.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::2cdd:6aff:fe73:ba69/64 scope link 
       valid_lft forever preferred_lft forever

```